mesonbuild/wrap/wrap.py

```diff
@@ -566,18 +566,21 @@ def get_file_internal(self, what: str) -> str:
 
             return path.as_posix()
 
+    def unpack_archive(self, path):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            shutil.unpack_archive(path, tmpdir)
+            src = os.listdir(tmpdir)
+            if len(src) != 1:
+                raise WrapException("Error patch archive format.")
+            self.copy_tree(src[0], os.path.join(self.subdir_root, self.wrap.name))
```

Comment:
Probably, `self.wrap.name` always contains name obtainted from `.wrap` filename without version suffix like `foo-1.0` and this causes CI fail in case whan wrap filename isn't same as directory of downloaded subproject

Reply:
Correct, the `{name}.wrap` is probably never (or rarely) the same as the `directory=`